CMakeLists.txt

```diff
@@ -0,0 +1,93 @@
+cmake_minimum_required(VERSION 3.25)
+
+project(fins LANGUAGES C VERSION 1.0)
+
+# compiler settings
+if(MSVC)
+    set(
+        COMPILER_C_FLAGS
+        -Ot
+        -GT
+        -DWINVER=0x0501
+        -D_WIN32_WINNT=0x0501
+        -volatile:iso
+        -nologo
+        -J
+        -sdl
+        -Wall
+        -WX
+        -wd4464
+        -wd4668
+        -wd4710
+        -wd4711
+        -wd4201
+        -wd4820
+        -Wv:18
+    )
+else()
+    set(
+        COMPILER_C_FLAGS
+        -Wall
+        -Wstrict-prototypes
+        -Wshadow
+        -Wpointer-arith
+        -Wcast-qual
+        -Wcast-align
+        -Wwrite-strings
+        -Wnested-externs
+        -Wredundant-decls
+        -Werror
```

Comment:
Do not generally add `-Werror`. It is fine for developers, CI and controlled platforms. It is a barrier for new platforms, new compiler versions and for users of build recipes (such as vcpkg ports).